addons/base/utils.py

```diff
@@ -54,3 +55,68 @@ def format_last_known_metadata(auth, node, file, error_type):
         ]
         return ''.join(parts)
     return msg
+
+
+class GravyValetAddonAppConfig:
+    class MockNodeSetting:
+        def __init__(self, resource, auth, legacy_config, gv_data):
```

Comment:
Something like:
```
        def __init__(self, configured_storage_addon_id):
           self.gv_data = requests.get(configured-storage-addons/configured-storage-addon-id/?include=external_storage_service)
           self.legacy_app_config = _get_legacy_config_for_gv_service(gv_data.name)
           self.magically_mirror_legacy_node_settings(legacy_app_config.node_settings_module)
           
       @property
       def config:
          return self.legacy_app_config
```

"magically_mirror_legacy_node_settings" could come from a per-provider translator class/module that is surfaced with a standardized name on the `legacy_config`